core/src/trezor/ui/layouts/mercury/__init__.py

```diff
@@ -236,6 +264,19 @@ async def raise_if_not_confirmed(
         raise exc
 
 
+async def interact(
+    layout: LayoutType[T],
+    br_type: str,
+    br_code: ButtonRequestType = ButtonRequestType.Other,
+) -> T:
+    pages = None
+    if hasattr(layout, "page_count") and layout.page_count() > 1:  # type: ignore [Cannot access member "page_count" for type "LayoutType"]
+        # We know for certain how many pages the layout will have
+        pages = layout.page_count()  # type: ignore [Cannot access member "page_count" for type "LayoutType"]
+    await button_request(br_type, br_code, pages)
+    return await layout
```

Comment:
This function is now duplicated with the only difference that `context.wait(...)` is not used. Thinking if it does not have unexpected consequences :thinking:

Reply:
Yes, it's pointless to have it duplicated, this should be consistent for all models really. I believe in #3686 the layout implementations are unified anyway.

I wanted to get some feedback from @matejcik regarding the python side of the PR before rewriting everything, but maybe I should port it to `main` for all models and see if it breaks any tests first.